From e35c4f412a3ca48245ed15ef7d00f1402d60e297 Mon Sep 17 00:00:00 2001
From: Jun Li <r65092@freescale.com>
Date: Mon, 15 Jun 2009 10:53:16 +0800
Subject: [PATCH] ENGR00113331 set host mode after usbcmd reset for ehci_run.

For those designs that contain both host & device capability,
the controller will default to an idle state and will need to
be initialized to the desired operating mode after reset.
For combination host/device controllers of FSL, SW need program
USBMODE register to make controller to be in host mode.

Signed-off-by: Li Jun <r65092@freescale.com>
---
 drivers/usb/host/ehci-hcd.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/host/ehci-hcd.c b/drivers/usb/host/ehci-hcd.c
index 058219e..58bd08e 100644
--- a/drivers/usb/host/ehci-hcd.c
+++ b/drivers/usb/host/ehci-hcd.c
@@ -594,6 +594,14 @@ static int ehci_run (struct usb_hcd *hcd)
 #endif
 	}
 
+	/* For those designs that contain both host & device capability,
+	 * the controller will default to an idle state and will need to
+	 * be initialized to the desired operating mode after reset.
+	 * For combination host/device controllers of FSL, SW need program
+	 * For combination host/device controllers of FSL, SW need program
+	 */
+	temp = readl(hcd->regs + 0x1a8);
+	writel(temp | (3 << 0), hcd->regs + 0x1a8);
 
 	// Philips, Intel, and maybe others need CMD_RUN before the
 	// root hub will detect new devices (why?); NEC doesn't
-- 
1.5.4.4

